site/src/contexts/ProxyContext.tsx

```diff
@@ -0,0 +1,185 @@
+import { useQuery } from "@tanstack/react-query"
+import { getApplicationsHost, getWorkspaceProxies } from "api/api"
+import { Region } from "api/typesGenerated"
+import { useDashboard } from "components/Dashboard/DashboardProvider"
+import { createContext, FC, PropsWithChildren, useContext, useState } from "react"
+
+interface ProxyContextValue {
+  proxy: PreferredProxy
+  isLoading: boolean
+  error?: Error | unknown
+  setProxy: (regions: Region[], selectedRegion: Region | undefined) => void
+}
+
+interface PreferredProxy {
+  // SelectedRegion is the region the user has selected.
+  // Do not use the fields 'path_app_url' or 'wildcard_hostname' from this
+  // object. Use the preferred fields.
+  selectedRegion: Region | undefined
+  // PreferredPathAppURL is the URL of the proxy or it is the empty string
+  // to indicte using relative paths. To add a path to this:
+  //  PreferredPathAppURL + "/path/to/app"
+  preferredPathAppURL: string
+  // PreferredWildcardHostname is a hostname that includes a wildcard.
+  preferredWildcardHostname: string
+}
+
+const ProxyContext = createContext<ProxyContextValue | undefined>(undefined)
+
+/**
+ * ProxyProvider interacts with local storage to indicate the preferred workspace proxy.
+ */
+export const ProxyProvider: FC<PropsWithChildren> = ({ children }) => {
```

Comment:
IMO this provider should lazily load all regions and have a `loading: bool` field. We can use the existing value from localStorage verbatim until we've loaded the regions list and can update it with a server-approved value.

Reply:
I don't feel a huge need to optimize. I removed the previous application host loading which was used in the workspace dashboard and terminal.

So optimizing this to lazy load would save a http call if the user would never hit the settings or workspace page.

I'd rather just be up front and make the call for now.